hotel-service/src/main/java/edu/myhotel/hotel/service/services/SearchService.java

```diff
@@ -22,4 +23,24 @@ public SearchResult searchAvailableRoom(LocalDate checkIn, LocalDate checkOut) {
         List<Room> availableRooms = dataStorage.findAvailableRooms(checkIn, checkOut);
         return new SearchResult(availableRooms, checkIn, checkOut);
     }
+
+    public RoomDetailInfo getRoomDetailInfo(int roomId) {
+        if (roomId < 1 || roomId > 7) {
+            throw new RoomNotFoundException("Room not found for id = " + roomId);
+        }
```

Comment:
Please check result of `dataStorage.getRoomById(roomId)` for throwing the error instead of using hardcoded check.